ProcessLib/Output.cpp

```diff
@@ -134,35 +134,30 @@ void Output::addProcess(ProcessLib::Process const& process,
                                  std::forward_as_tuple(filename));
 }
 
-Output::SingleProcessData Output::findSingleProcessData(
-    Process const& process, const unsigned process_id) const
+Output::SingleProcessData* Output::findSingleProcessData(
+    Process const& process, const int process_id)
 {
-    if (process.isMonolithicSchemeUsed())
-    {
-        auto spd_it = _single_process_data.find(&process);
-        if (spd_it == _single_process_data.end()) {
-            OGS_FATAL("The given process is not contained in the output"
-                      " configuration. Aborting.");
-        }
-        return spd_it->second;
-    }
-
     auto spd_range = _single_process_data.equal_range(&process);
-    unsigned counter = 0;
+    int counter = 0;
+    SingleProcessData* spd_ptr = nullptr;
     for (auto spd_it=spd_range.first; spd_it!=spd_range.second; ++spd_it)
```

Comment:
There is `safe_advance` solution on SO https://stackoverflow.com/a/1057782 .
Could be implemented in BaseLib or be done later.

Reply:
This loop will be remove after removing `Output::SingleProcessData`, which will be presented in another PR.